openpype/plugins/publish/collect_anatomy_instance_data.py

```diff
@@ -193,13 +195,25 @@ def fill_anatomy_data(self, context):
                 version_number = instance.data.get("version")
             # If version is not specified for instance or context
             if version_number is None:
-                # TODO we should be able to change default version by studio
-                # preferences (like start with version number `0`)
-                version_number = 1
+                task_name = instance.data.get("task", "")
```

Comment:
You could move version fill part after other data are filled, so you can just reuse them without own collection (speaking about task name and task type).

Reply:
This was not resolved, there is the same logic twice now in single scope (https://github.com/ynput/OpenPype/pull/5262/files#diff-6e9a139f3c7cd9ccb9efe39ac9c5f56b0d2547c1877af584ee34f704d0416356R242 ).

Reply:
Just move the `version_number` logic under the context logic.